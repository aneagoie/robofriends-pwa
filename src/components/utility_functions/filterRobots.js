// utils/filterRobots.js
export const filterRobots = (robots, searchField='') => {
    return robots.filter(robot =>
      robot.name.toLowerCase().includes(searchField.toLowerCase())
    );
  };
  