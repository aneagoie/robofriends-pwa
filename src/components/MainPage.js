import React, { Component } from 'react';
import CardList from '../components/CardList';
import SearchBox from '../components/SearchBox';
import Scroll from '../components/Scroll';
import ErrorBoundry from '../components/ErrorBoundry';
import Header from '../components/Header';
import { filterRobots } from '../components/utility_functions/filterRobots'; 
import './MainPage.css';


class MainPage extends Component {
 
  componentDidMount() {
    this.props.onRequestRobots();
  }

  render() {
    const { robots, searchField, onSearchChange, isPending } = this.props;
   
    return (
      <div className='tc'>
        <Header />
        <SearchBox searchChange={onSearchChange}/>
        <Scroll>
          { isPending ? <h1>Loading</h1> :
            <ErrorBoundry>
              <CardList robots={filterRobots(robots)} />
            </ErrorBoundry>
          }
        </Scroll>
      </div>
    );
  }
}

export default MainPage
