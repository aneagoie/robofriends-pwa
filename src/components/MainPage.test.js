import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import MainPage from "./MainPage";
import userEvent from '@testing-library/user-event';
import { Provider } from 'react-redux';
import { filterRobots } from './utility_functions/filterRobots';

let wrapper;

const mockProps = {
    onRequestRobots: jest.fn(),
    robots: [{
        id: 3,
        name: 'John',
        email: 'john@gmail.com'
    }],
    searchField: '',
    isPending: false,
};
beforeEach(() => {
    wrapper = render(<MainPage {...mockProps} />);
});

it("renders the MainPage component without crashing", () => {
    const { asFragment } = wrapper;
    expect(asFragment()).toMatchSnapshot();
})

it('filters robots correctly', () => {
 expect(filterRobots(mockProps.robots, '')).toEqual([{
    id: 3,
    name: 'John',
    email: 'john@gmail.com'
}]);
})