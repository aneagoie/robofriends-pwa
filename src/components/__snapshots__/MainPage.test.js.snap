// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the MainPage component without crashing 1`] = `
<DocumentFragment>
  <div
    class="tc"
  >
    <div>
      <h1
        class="f1"
      >
        RoboFriends
      </h1>
      <button
        data-testid="counter"
      >
        Count: 1
      </button>
    </div>
    <div
      class="pa2"
    >
      <input
        aria-label="Search"
        class="pa3 ba b--green bg-lightest-blue"
        placeholder="search robots"
        type="search"
      />
    </div>
    <div
      style="overflow: scroll; border: 5px solid black; height: 800px;"
    >
      <div>
        <div
          class="tc grow bg-light-green br3 pa3 ma2 dib bw2 shadow-5"
        >
          <img
            alt="robots"
            src="https://robohash.org/3?size=200x200"
          />
          <div>
            <h2>
              John
            </h2>
            <p>
              john@gmail.com
            </p>
          </div>
        </div>
        <div
          class="tc grow bg-light-green br3 pa3 ma2 dib bw2 shadow-5"
        >
          <img
            alt="robots"
            src="https://robohash.org/4?size=200x200"
          />
          <div>
            <h2>
              Alex
            </h2>
            <p>
              alex@gmail.com
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
